test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +112,105 @@ func testManualMeteringInstall(
 	assert.NoError(t, err, "capturing logs and uninstalling metering should produce no error")
 }
 
+func createNFSProvisioner(ctx *deployframework.DeployerCtx) error {
+	//Waiting for Metering pods
+	ctx.TargetPodsCount = nonHDFSTargetPodCount
+
+	_, err := createTestingNamespace(ctx.Client, "nfs")
+	if err != nil {
+		return err
+	}
+
+	files := map[string]string{
+		"pod":          "/test/testhelpers/nfs_server.yaml",
+		"service":      "/test/testhelpers/nfs_service.yaml",
+		"storageClass": "/test/testhelpers/nfs_storageclass.yaml",
+
+	}
+	server := &corev1.Pod{}
+	service := &corev1.Service{}
+	storageClass := &v1beta1.StorageClass{}
+
+	for name, file := range files {
+
+		switch name {
+		case "pod":
+			if err := deploy.DecodeYAMLManifestToObject(file, server); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Pods("nfs").Create(context.TODO(), server, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "service":
+			if err := deploy.DecodeYAMLManifestToObject(file, service); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Services("nfs").Create(context.TODO(), service, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "storageClass":
+			if err := deploy.DecodeYAMLManifestToObject(file, storageClass); err != nil {
+				return err
+			}
+			ctx.Client.StorageV1beta1().StorageClasses().Create(context.TODO(), storageClass, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+	}
+	// wait and Poll , get service
+	svc, err := ctx.Client.CoreV1().Services("nfs").Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+	if apierrors.IsNotFound(err) && svc.Spec.ClusterIP == ""{
+
+		err = wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
+
+			svc, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+
+			return svc.Spec.ClusterIP != "", err
+
+		})
+
+	}
+
+	nfsFile := "/test/testhelpers/nfs_persistentvolume.yaml"
+	nfs_pv := &corev1.PersistentVolume{}
+	nfs_pv.Spec.NFS.Server = svc.Spec.ClusterIP
```

Comment:
This is the wrong order of operations here. Decode first, then update the `nfs_pv` structure once it's been populated by `deploy.DecodeYAMLManifestToObject`.

Reply:
Let's move this call to update the `.Spec.NFS.Server` value to after lines 181-183 (after making the call to decode the YAML manifest into the `nfs_pv` empty object).